src/mutableAtom/index.ts

```diff
@@ -0,0 +1,155 @@
+import { Getter, Setter, atom } from 'jotai'
+import assert from 'minimalistic-assert'
+import { proxy, snapshot, subscribe } from 'valtio'
+import {
+  Options,
+  PromiseOrValue,
+  ProxyRef,
+  ProxyState,
+  SetCb,
+  SetSelf,
+  Store,
+  Wrapped,
+  WriteFn,
+} from './types'
+
+export function mutableAtom<Value>(
+  value: Value,
+  options: Options<Value> = defaultOptions
+) {
+  const { proxyFn } = { ...defaultOptions, ...options }
+
+  const valueAtom = atom({ value })
+
+  const storeAtom = atom(
+    () => ({ unsubscribe: null, isMounted: false } as Store),
+    (get, set, writeFn: WriteFn) => writeFn(get, set)
+  )
+
+  if (process.env.NODE_ENV !== 'production') {
+    storeAtom.debugPrivate = true
+  }
+
+  storeAtom.onMount = (writeFn) => {
+    // switch to synchronous imperative updates on mount
+    writeFn((get, set) => createProxyState(get, (fn) => fn(set)))
+    return () => writeFn(onAtomUnmount)
+  }
+  /**
+   * unsubscribe on atom unmount
+   */
+  function onAtomUnmount(get: Getter) {
+    get(storeAtom).unsubscribe?.()
+    get(storeAtom).isMounted = false
+  }
+
+  const proxyRefAtom = atom(() => ({ current: null } as ProxyRef<Value>))
+  if (process.env.NODE_ENV !== 'production') {
+    proxyRefAtom.debugPrivate = true
```

Comment:
Do we not mark as private for `storeAtom`?

Reply:
Its marked private on line 30

Reply:
oops, missed that.